### DISCUSSION TITLE
Caracal and subtracting point sources in galaxy clusters observations

### DISCUSSION THREAD

**Discussion by giannibernardi** [caracal-pipeline/caracal#1332] [General]
Radio observations of galaxy clusters often reveal compact (i.e. galaxy) as well as diffuse (halos and relics) emission. Point sources are often embedded in the diffuse emission (or sometimes sufficiently nearby) so that an unbiased observation of radio halos and relics require their subtraction. A standard procedure is the following:

- after selfcalibration has converged, make an image with only long baselines to filter our diffuse emission (possibly the halo, if already visible). This image should then be used to create the point source model (unless already available otherwise). There is no clear definition of what "long baselines" are and it depends upon the specific observation;

- subtract the point source model from visibilities and image the residual visibilities. As the goal of this residual image is to increase the halo/relic SNR, it is normally obtained by degrading the resolution (either playing with the robust parameter or chopping off long baselines or both). Also in this case there is not a unique recipe, but the final choice is what eventually maximizes the diffuse emission SNR and still places a reasonable number of PSFs across the halo/relic.

At that point measurements of the halo flux density, spectral index maps etc... can be safely derived. (the above procedure may need a few iterations before converging to the best result.

**Comment by viralp** [comment]
@Gianni Bernardi ***@***.***>
Thanks for your recipe. I would like to add here that we have tested this
procedure of point source subtraction with the crystalball. You need to
select point sources using the ds9, that you want to subtract from the
visibility. Then give this ds9 region file and wsclean generated clean
components list to the crystalball. Then crystalball will create another
column in the ms file, which has only clean components of point sources.
Finally subtract this column from the corrected_data and reimage the
residual column. We have tested this method for MeerKAT and it is quite
well working. See the below an example image where we subtracted bright
compact sources.

[image: image.png]



On Fri, May 14, 2021 at 3:30 PM giannibernardi ***@***.***>
wrote:

> Radio observations of galaxy clusters often reveal compact (i.e. galaxy)
> as well as diffuse (halos and relics) emission. Point sources are often
> embedded in the diffuse emission (or sometimes sufficiently nearby) so that
> an unbiased observation of radio halos and relics require their
> subtraction. A standard procedure is the following:
>
>    -
>
>    after selfcalibration has converged, make an image with only long
>    baselines to filter our diffuse emission (possibly the halo, if already
>    visible). This image should then be used to create the point source model
>    (unless already available otherwise). There is no clear definition of what
>    "long baselines" are and it depends upon the specific observation;
>    -
>
>    subtract the point source model from visibilities and image the
>    residual visibilities. As the goal of this residual image is to increase
>    the halo/relic SNR, it is normally obtained by degrading the resolution
>    (either playing with the robust parameter or chopping off long baselines or
>    both). Also in this case there is not a unique recipe, but the final choice
>    is what eventually maximizes the diffuse emission SNR and still places a
>    reasonable number of PSFs across the halo/relic.
>
> At that point measurements of the halo flux density, spectral index maps
> etc... can be safely derived. (the above procedure may need a few
> iterations before converging to the best result.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/caracal-pipeline/caracal/discussions/1332>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABBBM4YMTNWLIQZUBWE3K3DTNUQXJANCNFSM444ODJYQ>
> .
>


-- 
Postdoctoral Research Fellow
South African Radio Astronomy Observatory (SARAO) and Rhodes University
Square Kilometre Array, Cape Town
South Africa

**Reply by o-smirnov** [reply]
@viralp thanks, but could you please upload the image on the github discussion page? It doesn't look like it is capable of including images by e-mail.

**Reply by viralp** [reply]
Here is the example image of one galaxy cluster after subtraction of point sources. 

![image](https://user-images.githubusercontent.com/4331123/118303091-561a3280-b4e5-11eb-84df-95d36fa4f7e4.png)

**Comment by giannibernardi** [comment]
@viralp thanks! The goal in opening this thread was to check if an effective recipe could be pipelined (or is already pipelined)... I am ignorant of crystalball, but your residual image looks good - although it seems there is residual left at the source location for many sources after subtraction... Is there diffuse emission in the right image?

**Reply by viralp** [reply]
@giannibernardi we do have a stimela script to run this task which canbe include in a pipeline. No in this cluster, we haven't detected any diffuse emission.

**Reply by o-smirnov** [reply]
So I think you're discussing two orthogonal approaches, which could be (I'm guessing fruitfully!) combined:

* Crystallball does is a DFT on the clean component list generated by wsclean. This is more accurate than using model images, because it is able to predict a smooth spectral polynomial at native channel resolution.

* @giannibernardi is proposing a uv-cut to image with long baselines only, thus forcing the clean model to contain only compact sources.

Why not both at once?

**Comment by giannibernardi** [comment]
both at once sounds good to me. The only thing that I would pay attention to (although not crucial, perhaps) is the depth of the component list - i.e. ry to do a reasonably deep clean in order to model source flux densitiy accurately. In principle also a PyBDSF model would be appropriate, but if Crystalball is already impemente in the pipeline it would make sense to start with it. @o-smirnov: our A141 images have a couple of point sources embedded in diffuse emission...

**Comment by viralp** [comment]
Yes I also agree to combine both and have tried it. First make a uniform
image (or robust weighting -2) using WSClean, select the sources to be
subtracted in the ds9 and supply both the clean component list and region
file to crystalball.

On Sat, May 15, 2021 at 11:07 PM Oleg Smirnov ***@***.***>
wrote:

> So I think you're discussing two orthogonal approaches, which could be
> (I'm guessing fruitfully!) combined:
>
>    -
>
>    Crystallball does is a DFT on the clean component list generated by
>    wsclean. This is more accurate than using model images, because it is able
>    to predict a smooth spectral polynomial at native channel resolution.
>    -
>
>    @giannibernardi <https://github.com/giannibernardi> is proposing a
>    uv-cut to image with long baselines only, thus forcing the clean model to
>    contain only compact sources.
>
> Why not both at once?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/caracal-pipeline/caracal/discussions/1332#discussioncomment-742816>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABBBM44UGFQN3CB5UGQ3JSDTN3PBVANCNFSM444ODJYQ>
> .
>


-- 
Postdoctoral Research Fellow
South African Radio Astronomy Observatory (SARAO) and Rhodes University
Square Kilometre Array, Cape Town
South Africa

**Reply by giannibernardi** [reply]
well, probably is more than a uniform image... it is a uniform image after a uv cut... but let's take a step at the time, thanks!

**Comment by paoloserra** [comment]
Hi all,

Thanks for starting this interesting discussion @giannibernardi . I think what you suggest is already possible in CARAcal with an ad hoc strategy, which could look something like this:

```
# usual imaging=selfcal loop on all baselines, after which CORRECTED and MODEL contain all sources
selfcal__all:
  <first iteration images DATA, following iterations image CORRECTED>

# one round of imaging of CORRECTED using long baselines only -- no calibration here.
selfcal__ps:
  minuvw_m: <some values>
  <and all other options, making sure you image CORRECTED and not DATA>

# overwrite CORRECTED with CORRECTED - MODEL
line__subtractps:
  subtractmodelcol:
    enable: true
  <everything else disabled>

# new imaging of diffuse emission only -- no calibration here.
selfcal__diffuse:
  <some imaging settings, making sure that you image CORRECTED and not DATA>
```

Having to use the `line` worker to subtract the clean components is not particularly elegant, but well...

Another thing to be careful about is the overwriting of CORRECTED with CORRECTED-MODEL. It's nasty. Maybe make a backup of the MS.

I suspect that  this  strategy is faster and less painful than one where you need to find and identify  with a ds9 region all point sources in  the field and overwrite the MODEL column with crystalball.

**Comment by ifornax** [comment]
This paper may be useful that Cyril used KillMS to detect the halo
https://repository.nwu.ac.za/bitstream/handle/10394/18420/KAT-7%20detection.pdf?sequence=1
